stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -373,12 +373,12 @@ SparseMatrixCSC(M::AbstractMatrix{Tv}) where {Tv} = SparseMatrixCSC{Tv,Int}(M)
 SparseMatrixCSC{Tv}(M::AbstractMatrix{Tv}) where {Tv} = SparseMatrixCSC{Tv,Int}(M)
 function SparseMatrixCSC{Tv,Ti}(M::AbstractMatrix) where {Tv,Ti}
     @assert !has_offset_axes(M)
-    I = findall(x -> x != 0, M)
-    eltypeTiI = Ti[i[1] for i in I]
-    eltypeTiJ = Ti[i[2] for i in I]
-    eltypeTvV = Tv[M[i] for i in I]
-    return sparse_IJ_sorted!(eltypeTiI, eltypeTiJ, eltypeTvV, size(M)...)
+    C = findall(v->v != 0, M)
+    @inbounds I = convert(Vector{Ti},LinearIndices(M)[C])
```

Comment:
Do these `@inbounds` do anything?

Reply:
The first two yes (a bit less than 10% less time with n=1000, p=0.1). The last one not really, I've removed it. Thanks @KristofferC.

Reply:
This seems like it could make two copies of the index array.  Probably better to do:
```jl
Mi = LinearIndices(M)
@inbounds I = Ti[Mi[c] for c in C]
```

Reply:
Actually, better to get rid of the `C` array entirely:
```jl
Mi = LinearIndices(M)
@inbounds I = Ti[Mi[first(p)] for p in pairs(M) if !iszero(last(p))]
```